services/workflows-service/src/workflow/workflow-definition.repository.ts

```diff
@@ -45,6 +45,19 @@ export class WorkflowDefinitionRepository {
     );
   }
 
+  async findTemplateByIdUnscoped<
+    T extends Omit<Prisma.WorkflowDefinitionFindFirstOrThrowArgs, 'where'>,
+  >(
+    id: string,
+    args: Prisma.SelectSubset<T, Omit<Prisma.WorkflowDefinitionFindFirstOrThrowArgs, 'where'>>,
+    projectId?: string,
+  ): Promise<WorkflowDefinition> {
+    return await this.prisma.workflowDefinition.findFirstOrThrow({
+      where: { id, isTemplate: true, projectId: projectId ?? null },
```

Comment:
Remove `projectId` as it's irrelevant here.
```suggestion
  ): Promise<WorkflowDefinition> {
    return await this.prisma.workflowDefinition.findFirstOrThrow({
      where: { id, isTemplate: true },
```

Reply:
I somewhat agree.
but there's some kind of logic of creating "templates" (publics) to specific customers.
e.g. nuvei will have multiple of the same template